packages/ember-data/lib/system/model/model.js

```diff
@@ -690,18 +704,20 @@ var Model = Ember.Object.extend(Ember.Evented, {
   */
   clearRelationships: function() {
     this.eachRelationship(function(name, relationship) {
-      var rel = this._relationships[name];
-      if (rel) {
-        //TODO(Igor) figure out whether we want to clear or disconnect
-        rel.clear();
-        rel.destroy();
+      if (this._relationships.has(name)) {
+        var rel = this._relationships.get(name);
+        if (rel) {
```

Comment:
Now that we have the `has` check do we need this second test on the `rel` value?